retirement-ui/src/api/solanaRetirement.ts

```diff
@@ -187,25 +205,39 @@ export class SolanaRetirement {
             true
         );
         console.log(`ata: ${ata.toBase58()}`);
-        
-        let ataCreationTx = new Transaction();
-        ataCreationTx.add(
-            createAssociatedTokenAccountInstruction(
-            this.solWallet.publicKey,
-            ata,
-            tokenAuthority,
-            BRIDGE_INPUT_MINT_ADDRESS
-            )
-        );
+
+        const ataAccountInfo = await this.solConnection.getAccountInfo(ata)
         let preIxs: TransactionInstruction[];
-        if (preInstructions?.length == 0) {
-            preIxs = ataCreationTx.instructions;
+        if (ataAccountInfo == null) {
+        
+            let ataCreationTx = new Transaction();
+            ataCreationTx.add(
+                createAssociatedTokenAccountInstruction(
+                this.solWallet.publicKey,
+                ata,
+                tokenAuthority,
+                BRIDGE_INPUT_MINT_ADDRESS
+                )
+            );
+            if (preInstructions?.length == 0) {
```

Comment:
Makes sense, but could be neater I feel. How about something along the lines of?

```suggestion
           let preIxes = []; // this goes above - line 210
           ...
           preIxs.push(...preInstructions ?? []) // push anything from preInstructions if they exist
           preIxs.push(...ataCreationTx.instructions)
```

you could also move the preinstructions push step above the if statement since it is added in either case, then remove the lines 232-240 too.